Port update package version improvements (#23)
valory-xyz/open-aea: scripts/RELEASE_PROCESS.md

# Release Process from develop to main

1. Make sure all tests pass, coverage is at 100% and the local branch is in a clean state (nothing to commit). Make sure you have a clean develop virtual environment. 
   
2. Determine the next AEA version 
   Create new release branch named "feature/release-{new-version}, switch to this branch"
   Run `python scripts/bump_aea_version.py --new-version NEW_VERSION_HERE`. Commit if satisfied.

3. Bump plugin versions if necessary by running `python scripts/update_plugin_versions.py --update "PLUGIN_NAME,NEW_VERSION"`. Commit if satisfied.

4. Check the protocols are up-to-date by running `python scripts/generate_all_protocols.py`. Commit if changes occurred.

> 5. Bump all the packages to their latest versions by running `python scripts/update_package_versions.py`.

Reviewer: please revert, we don't have this equivalent atm

13. Publish to PyPI with twine: `twine upload dist/*`. Optionally, publish to Test-PyPI with twine:
`twine upload --repository-url https://test.pypi.org/legacy/ dist/*`.

> 14. Repeat 11. & 12. for each plugin.

Reviewer: please revert

15. Make clean environment and install release from PyPI: `pip install aea[all] --no-cache`.

> 16. Release packages into registry: `python scripts/deploy_to_registry.py`. Run it several times till all packages updated, cause some dependencies order.

Reviewer: ```suggestion
16. [CURRENTLY SKIPPED] Release packages into registry: `python scripts/deploy_to_registry.py`. If necessary, run it several times until all packages are updated.
```

> 17. AEA develop docker image don automatically with CI with develop to main PR.

Reviewer: ```suggestion
17. The `user-image` and `develop-image` are update via CI.
```
